flink-connector/src/main/java/org/apache/flink/connector/delta/sink/SchemaConverter.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.delta.sink;
+
+import org.apache.flink.table.types.logical.LogicalType;
+import org.apache.flink.table.types.logical.RowType;
+
+import io.delta.standalone.types.*;
+
+/**
+ * This is a utility class to convert from Flink's specific {@link RowType} into
+ * DeltaLake's specific {@link StructType} which is used for schema-matching comparisons
+ * during {@link io.delta.standalone.DeltaLog} commits.
+ */
+public class SchemaConverter {
+
+    /**
+     * Main method for converting from {@link RowType} into {@link StructType}
+     *
+     * @param rowType Flink's logical type of stream's events
+     * @return DeltaLake's specific type of stream's events
+     */
+    public StructType toDeltaFormat(RowType rowType) {
+        StructField[] fields = rowType.getFields()
+                .stream()
+                .map(rowField -> {
+                    DataType rowFieldType = toDeltaDataType(rowField.getType());
+                    return new StructField(
+                            rowField.getName(),
+                            rowFieldType,
+                            rowField.getType().isNullable());
+                })
+                .toArray(StructField[]::new);
+
+        return new StructType(fields);
+    }
+
+    /**
+     * Method containing the actual mapping between Flink's and DeltaLake's types.
+     *
+     * @param flinkType Flink's logical type
+     * @return DeltaLake's data type
+     */
+    public DataType toDeltaDataType(LogicalType flinkType) {
```

Comment:
what is the difference between Flink's RowType and LogicalType? Basically, why is there a need for two very similar-looking functions?

Reply:
also... does this not handle nested types like StructType?

Reply:
i see that the RowType is equivalent to StructType.. right?

Reply:
yes, RowType is equivalent to StructType so it handles nested types. It seemed to me more natural to handle RowType conversions in a separate function, but if you think that we should keep it all in one function (with input parameter as LogicalType) then i can move the `toDeltaFormat` method's content under `case ROW:` switch statement.
what's your mind on this ?